gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+            method='PUT', path=self.path, data=self._to_dict())
+        self._init_from_dict(info)
+
+    def delete(self):
+        """Delete the group via a ``DELETE`` request.
+
+        Example::
+
+            >>> group = client.group('1234')
+            >>> group.delete()
+
+        Only the ``client`` and ``name`` attributes are used.
+
+        .. warning::
+
+            This method will fail for groups that have one or more children
+            groups.
+        """
+        self.client.connection.api_request(method='DELETE', path=self.path)
+
+    def parent(self):
+        """Returns the parent group of this group via a ``GET`` request.
+
+        :rtype: :class:`Group` or None
+        :returns: The parent of the group.
+        """
+        if not self.parent_id:
+            return None
+        return self._fetch(self.client, self.parent_id)
+
+    def children(self):
+        """Lists all children of this group via a ``GET`` request.
+
+        Returns groups whose parent_name field contains the group name. If no
+        groups have this parent, the results are empty.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, children_of_group=self.name)
+
+    def ancestors(self):
+        """Lists all ancestors of this group via a ``GET`` request.
+
+        The groups are returned in order, starting with the immediate parent
+        and ending with the most distant ancestor. If the specified group has
+        no immediate parent, the results are empty.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, ancestors_of_group=self.name)
+
+    def descendants(self):
+        """Lists all descendants of this group via a ``GET`` request.
+
+        This returns a superset of the results returned by the :meth:`children`
+        method, and includes children-of-children, and so forth.
+
+        :rtype: list of :class:`~gcloud.monitoring.group.Group`
+        :returns: A list of group instances.
+        """
+        return self._list(self.client, descendants_of_group=self.name)
+
+    def members(self, filter_string=None, end_time=None, start_time=None):
+        """Lists all members of this group via a ``GET`` request.
+
+        If no end_time and start_time is provided then the group membership
+        over the last minute is returned.
+
+        Examples::
+
+            To get current members that are Compute Engine VM instances:
+
+                group.members('resource.type = "gce_instance"')
+
+            To get historical members that existed between 4 and 5 hours ago:
+
+                import datetime
+                t1 = datetime.datetime.utcnow() - datetime.timedelta(hours=4)
+                t0 = t1 - datetime.timedelta(hours=1)
+                group.members(end_time=t1, start_time=t0)
```

Comment:
If you don't have `>>>` you'll want two colons `::` after ago to get highlighting. Why the switch here? All samples so far have used the console output (`>>>`).